src/node_report.cc

```diff
@@ -782,6 +792,38 @@ static void PrintRelease(JSONWriter* writer) {
   writer->json_objectend();
 }
 
+static void PrintRequestInfo(JSONWriter* writer,
+                             Environment* env,
+                             const char* trigger) {
+  bool printOwnerInfo = strcmp(trigger, "OOMError") != 0;
+  for (ReqWrapBase* req_wrap : *env->req_wrap_queue()) {
+    AsyncWrap* w = req_wrap->GetAsyncWrap();
+    if (w->persistent().IsEmpty()) continue;
+    writer->json_start();
+    writer->json_keyvalue("type", w->MemoryInfoName().c_str());
```

Comment:
We can get the type name of the uv request with `uv_req_type_name`

Reply:
Like this ?
```
ReqWrap *req_wrap = dynamic_cast<ReqWrap *>(req_wrap);
 if (req != nullptr)
 {
	 const char*  name = uv_req_type_name(req_wrap->req()->type);
 }
```

Reply:
I believe you can static_cast to `ReqWrapBase` and get the raw pointer of uv req with it.

Reply:
Do you mean `req()` method ? `req()` is in class `ReqWrap` but not in `ReqWrapBase` 🤔.

Reply:
In addition, I need to confirm whether the uv request struct has enough information. Thanks for your suggestion.